`wasmMemory.buffer instanceof SharedArrayBuffer === false` occasionally

Reproduction: 
- Open https://rolldown-browser.vercel.app/
- Typing some code without interruption

Source code: https://github.com/sxzz/rolldown-browser (cannot build locally)

```
index-DAfNCrBY.js:42708 Error: Build failed with 1 error:

TypeError: Failed to execute 'decode' on 'TextDecoder': The provided ArrayBufferView value must not be shared.
TypeError: The provided ArrayBufferView value must not be shared.
    at Object.UTF8ToString (index-DAfNCrBY.js:9305:49)
```

https://github.com/user-attachments/assets/c39733f3-d36b-4cbb-a2ac-01c82d81b648

/cc @brooooooklyn

---

Can we check buffer byusing `HEAPU8.buffer[Symbol.toStringTag] === 'SharedArrayBuffer'`?

--- a/packages/ts-transform-emscripten-parse-tools/src/index.ts
+++ b/packages/ts-transform-emscripten-parse-tools/src/index.ts
@@ -734,11 +734,8 @@ class Transform {
             ),
             undefined,
             [factory.createPropertyAccessExpression(
-              factory.createPropertyAccessExpression(
-                factory.createIdentifier(heap),
-                factory.createIdentifier('buffer')
-              ),
-              factory.createIdentifier('constructor')
+              factory.createIdentifier(heap),
+              factory.createIdentifier('buffer')
             )]
           ),
           factory.createToken(ts.SyntaxKind.EqualsEqualsEqualsToken),

--- a/packages/wasi-threads/src/util.ts
+++ b/packages/wasi-threads/src/util.ts
@@ -69,7 +69,7 @@ export function deserizeErrorFromBuffer (sab: SharedArrayBuffer): Error | null {
 export function isSharedArrayBuffer (value: any): value is SharedArrayBuffer {
   return (
     (typeof SharedArrayBuffer === 'function' && value instanceof SharedArrayBuffer) ||
-    (Object.prototype.toString.call(value.constructor) === '[object SharedArrayBuffer]')
+    (Object.prototype.toString.call(value) === '[object SharedArrayBuffer]')
   )
 }
 

